lib/jobs/wait-sel-events-job.js

```diff
@@ -0,0 +1,213 @@
+        })
+        .tap(function(pollers){
+            self._retrievePollerInfo(pollers);
+        })
+        .then(function(pollers){
+            return self.setSelAlertConfig(pollers[0]);
+        })
+        .then(function(){
+            return self._subscribeSelEvent(self.pollerId, self.nodeId, self._callback.bind(self));
+        })
+        .catch(function(err){
+            logger.error('Fail to run wait sel events', {
+                node: self.nodeId,
+                error: err,
+                options: self.options
+            });
+            //self._done(err);
+            self._done();
+        });
+    };
+
+    /**
+     * @memberOf waitSelEventJob
+     */
+    waitSelEventJob.prototype._retrievePollerInfo = function(pollers) {
+        var poller;
+        var existingAlerts;
+        var self = this;
+        if (_.isEmpty(pollers)) {
+            return Promise.reject("Can't find SEL poller");
+        }
+        if (pollers.length !== 1) {
+            return Promise.reject("Find more than one SEL poller for a node");
+        }
+        return Promise.try(function(){
+            poller = pollers[0];
+            self.pollerId = poller.id;
+            self.legacyInterval = poller.pollInterval;
+            existingAlerts = poller.config.alerts || [];
+            _.forEach(self.alertFilters, function(alertFilter){
+                var filter = _.omit(alertFilter, 'count');
+                if (!filter.action || filter.action !== 'critical'){
+                    filter.action = 'critical';
+                }
+                if (_.findIndex(existingAlerts, filter) === -1) {
+                    self.freshAlerts.push(filter); //filter.action should always be critical
+                }
+            });
+            self.pureFilters = _.map(self.alertFilters, function(alertFilter){
+                return _.omit(alertFilter, ['action', 'count']);
+            });
+            self.eventCounts = _.map(self.alertFilters, function(alertFilter){
+                return alertFilter.count || 1;
+            });
+        });
+    };
+
+    /**
+     * @memberOf waitSelEventJob
+     */
+    waitSelEventJob.prototype.updateSelPollerConfig = function(update) {
+        if (_.isEmpty(this.freshAlerts)){
+            update = _.omit(update, ['$addToSet', '$pullAll']) ;
+        }
+        if (this.legacyInterval === this.pollInterval) {
+            update = _.omit(update, '$set') ;
+        }
+        if (_.isEmpty(update)) {
+            return Promise.resolve();
+        } else if (!_.has(update, '$set')){
+            //To bypass waterline validation
+            update.$set = {};
+        }
+        var query = {_id: waterline.workitems.mongo.objectId(this.pollerId)};
+        return waterline.workitems.updateMongo(query, update, {});
+    };
+
+    /**
+     * @memberOf waitSelEventJob
+     */
+    waitSelEventJob.prototype.unsetSelAlertConfig = function() {
+        var self = this;
+        var update = {
```

Comment:
Previous there is a convention, the job should avoid to send raw database command, these commands should be moved into the on-core's waterline model.